test/src/helpers.cc

```diff
@@ -68,22 +73,111 @@ void check_partitions(
   for (const auto& p : partitions) {
     CHECK(!partitioner.done());
     CHECK(!unsplittable);
+    //TBD: *if* previous loop iteration next() happened to finish, following (next loop iteration) ...ok() will still return true,
+    //(as Status::Ok() is returned from top when already 'done()' on entry to next())
+    //tho suppose that partition check following (for actual data partition data) should fail, assuming the two last partition entries
+    //aren't the same - can the same partition information be returned more than once???
     CHECK(partitioner.next(&unsplittable).ok());
     auto partition = partitioner.current();
     check_subarray<T>(partition, p);
   }
 
   // Check last unsplittable
-  if (last_unsplittable) {
+  if (last_unsplittable) { //Was last partition/next() in prior expected to be unsplittable?
     CHECK(unsplittable);
   } else {
+    //TBD: but expectation here is that we did *not* pass all possible partitions in 'partitions', and hence
```

Comment:
i can't follow this, please rewrite for clarity into a succinct list of questions and bullet points